### DISCUSSION TITLE
v1.8.0 Release

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#577] [Announcements]
Release Notes
=========

## Breaking Change
- Plugin API (Flow.Launcher.Plugin) upgraded from .Net Core 3.1 to .Net 5

### Users: ###
If you are on a version prior to 1.8.0, you will no longer recieve plugin updates or newly added plugins. You can still continue to use your existing plugins and download all plugins released prior to Flow v1.8.0 release. 

We recommend you upgrade to v1.8.0 to continue to receive new plugins and updates, plus the below features and fixes.

### Plugin Developers: ###
We have added new functionalities in API v2.0.0 (.Net 5 requirement) which includes features such as saving settings via the API- you do not need to write your own code to save your plugin settings anymore.

## Features
General
- Install Flow via WinGet: `WinGet install "Flow Launcher"`
- New [documentation](https://flow-launcher.github.io/docs) page!
- Press F5 in query window to reload all plugin data
- Automatic Python installation for Python plugins 
- Added documentation usage tips link in the Settings About tab
- Optimized exception handling and logging
- Remove top and bottom margin of General tab (Settings) 
- Added the option to enable or disable auto hiding of the settings window scrollbar 
- Increased the Settings page's default width and allow user to resize the window

Performance
- Improved flow's download efficiency 
- Optimized result list queuing and updating, this means results return faster 
- Reworked result score bumping mechanism to allow more accurate score bumps for frequently used results

Plugin 
- Enable Flow's saving mechanism to be used by plugins via direct API call to save their settings
- Added logging, save and load settings json files to IPublicAPI for plugins to use 
- IPlugin interface now inherits from IAsyncPlugin 
- Optimize plugins manifest downloading behavior 
- Updated JSON-RPC model to async 
- Added copy command via context menu for Shell plugin
- Enabled saving of Shell plugin's commands history, with option to set the display number of most used commands in result list 
- Add API error msg handling method 
- Add tips, UserData and Log location commands for Sys plugin 
- Run Shell plugin cmds as admin with ctrl 
- Ctrl + enter/click opens Plugin's url in the plugins download list
- Copy the bookmark's url to clipboard via BrowserBookmarks's context menu 
- Multiple UX/UI enhancements to WebSearch plugin 
- Implemented the option to disable program description
- For Explorer plugin, in addition to the main search action keyword also added action keywords to separately search for directory paths and indexed files and folders 
- Allow Quick Access Links to be searched using part of the string instead of the word starting with 

## Bug fixes
General
- Fixed issue causing Flow to turn into  a vertical black line after waking from hibernation 
- Fixed the Settings window freezing after locking computer screen (Win+L) 
- Fixed some minor debuging issues
- Fixed query window's shadow margin not appearing at top and bottom, as well as being cut off at the corners 
- Fixed Context Menu bug where when no item is selected going to the context menu results in weird UX 
- Fixed the Query Box Is Not Focused When HotKey Invoked when using Shell plugin's Win+R shortcut 
- Fix flow's hotkey not working correctly or responding after changing in settings 

Performance
- Fixed progress bar animation, which should drop cpu usage when hidden while query is running 
- Fixed settings window doesn't show back up when minimized in taskbar

Plugins
- Fixed win32 app loading for Program plugin: 
  - Now able to retrieve the correct file path for programs in the start menu that have LNK extension
  - Fix the issue where type 'cmd' does not get you Command Prompt
  - Remove ShObjldITIb.dll used for Win32 loading, we now directly use the code
  - Remove duplicate lnk program pointing to the same exe with the same description
- Removed AppxPackagingTlb.dll used for UWP loading 
- Fixed a bug related to operation timeout issue for Flow's version updater and WebSearch plugin 
- Updated json response usage in WebSearch plugin
- Fixed WebSearch plugin's action keyword: 
  - Changed the WebSearch's google search action keyword to global as the default 
  - Fixed ordering of the plugin's result so non global action keyword matches will appear above global action keyword matches
- Re-added missing Google Scholar action keyword `sc` 
- Fixed the issue where indexing would need to finish before Flow could continue loading the rest of startup 
- Fixed exclusion of Windows Index search, making it effective also for searches via file/folder names or file contents 
- Optimized BrowserBookmarks and Calculator plugins' startup speed 
- Remove redundant QuickAccess folder nickname path 
- Added/updated Slovak translation for plugins 
- Added and corrected Chinese translation 
- Fix Plugin Indicator when more than one query term is typed 
- Fixed Win32 & UWP's AppxManifest.xml leak 
- Fixed loading when no action keywords present for a plugin in user plugin setting 
- Fixed duplicate query searches from same global action keywords 
- Fix unparsed property when using Python plugins 
- Changed sequence of loading for Shell plugin to avoid null reference 
- Fixed JSON-RPC model issues 
- Fixed a potential issue with querying after loading a Python context menu 
- Fixed PluginsManifest constantly updating in the background when queried 
- Removed empty entries when specifying program suffixes 
- Fixed missing disposable call that allows plugins to do something before flow exits 

<hr /><em>This discussion was created from the release <a href='https://github.com/Flow-Launcher/Flow.Launcher/releases/tag/v1.8.0'>v1.8.0</a>.</em>

**Comment by kubalav** [comment]
It seems wrong elements alignment.

![wrong-alignment](https://user-images.githubusercontent.com/37414585/126108015-4acc2989-e795-4678-896a-b71044e55e47.png)

**Reply by taooceros** [reply]
You mean it aligns too low?

**Reply by kubalav** [reply]
Yes, I do. Drop-down menu should be higher.

**Reply by taooceros** [reply]
#579

**Reply by kubalav** [reply]
Thanks. Maybe move drop-down menu & textbox like that:

![wrong-alignment-suggestion](https://user-images.githubusercontent.com/37414585/126127833-b9b75f74-168a-43bb-84a2-909eb1939126.png)

**Reply by taooceros** [reply]
Oh got you. Will take a try later.
